yeat/cli/__init__.py

```diff
@@ -0,0 +1,58 @@
+# -------------------------------------------------------------------------------------------------
+# Copyright (c) 2021, DHS. This file is part of YEAT: http://github.com/bioforensics/yeat
+#
+# This software was prepared for the Department of Homeland Security (DHS) by the Battelle National
+# Biodefense Institute, LLC (BNBI) as part of contract HSHQDC-15-C-00064 to manage and operate the
+# National Biodefense Analysis and Countermeasures Center (NBACC), a Federally Funded Research and
+# Development Center.
+# -------------------------------------------------------------------------------------------------
+
+from .cli import get_parser
+from pathlib import Path
+from pkg_resources import resource_filename
+from snakemake import snakemake
+from yeat.assembly import config
+
+
+SNAKEFILES = {"spades": resource_filename("yeat", "assembly/Snakefile_Spades")}
+
+
+def run(read1, read2, assembler, outdir=".", cores=1, sample="sample", dryrun="dry"):
+    snakefile = SNAKEFILES[assembler.algorithm]
```

Comment:
Each assembly will have its' own unique snake file. 

I decided to go with this route because I wanted to each assembly job to have their own DAG. In the Snakemake file, it is possible to go with this route: 

```
if "spades": 
    rule spades:
        blah...

if "megahit":
    rule megahit:
        blah...
```

However, by having these branching statements, there will be a problem of conditional generated DAGs. Maybe it's okay to have this, but I think having a complete DAG job of each assembly would be better than an incomplete DAG with branching complete DAG jobs based on conditions.

Reply:
![image](https://user-images.githubusercontent.com/33472323/160459402-2b55d324-9398-4c60-b71f-f213287a7957.png)